tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/CompileResult.java

```diff
@@ -34,27 +32,25 @@
 public class CompileResult {
 
     private PackageNode pkgNode;
-    private CompileResultDiagnosticListener diagnosticListener;
 
     URLClassLoader classLoader;
 
-    public CompileResult(CompileResultDiagnosticListener diagnosticListener) {
-        this.diagnosticListener = diagnosticListener;
+    public CompileResult() {
     }
 
     public Diagnostic[] getDiagnostics() {
-        List<Diagnostic> diagnostics = this.diagnosticListener.getDiagnostics();
-        diagnostics.sort(Comparator.comparing((Diagnostic d) -> d.getSource().getCompilationUnitName()).
-                thenComparingInt(d -> d.getPosition().getStartLine()));
+        List<Diagnostic> diagnostics = ((BLangPackage) this.pkgNode).getDiagnostics();
+        diagnostics.sort(Comparator.comparing((Diagnostic d) -> d.location().lineRange().filePath()).
```

Comment:
Shall we use an ordered list in the bLangPackage to store diagnostics? Then we don't have to sort it everytime..

Reply:
Shouldn't we have to implement `comparedTo` methods ?

Reply:
+1 for using an ordered list. Sorting every time can be a burden when we are doing a migration between versions where there are considerable amount of diagnostics.